### PR TITLE
Update to latest substrate

### PR DESCRIPTION
**Pull request by tomusdrw**
Supersedes #722

### CODE REVIEW

File: modules/substrate/Cargo.toml [hunk 1]
```diff
@@ -8,8 +8,8 @@ license = "GPL-3.0-or-later WITH Classpath-exception-2.0"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-codec = { package = "parity-scale-codec", version = "1.3.1", default-features = false }
-finality-grandpa = { version = "0.12.3", default-features = false }
+codec = { package = "parity-scale-codec", version = "2.0.0", default-features = false }
+finality-grandpa = { version = "0.13.0", default-features = false }
```
Comment (adoerr) [inline modules/substrate/Cargo.toml:12]: And here.

File: primitives/header-chain/Cargo.toml [hunk 2]
```diff
@@ -7,8 +7,8 @@ edition = "2018"
 license = "GPL-3.0-or-later WITH Classpath-exception-2.0"
 
 [dependencies]
-codec = { package = "parity-scale-codec", version = "1.3.1", default-features = false }
-finality-grandpa = { version = "0.12.3", default-features = false }
+codec = { package = "parity-scale-codec", version = "2.0.0", default-features = false }
+finality-grandpa = { version = "0.13.0", default-features = false }
```
Comment (adoerr) [inline primitives/header-chain/Cargo.toml:11]: And here.

File: primitives/test-utils/Cargo.toml [hunk 3]
```diff
@@ -6,7 +6,7 @@ edition = "2018"
 license = "GPL-3.0-or-later WITH Classpath-exception-2.0"
 
 [dependencies]
-finality-grandpa = { version = "0.12.3" }
+finality-grandpa = { version = "0.13.0" }
```
Comment (adoerr) [inline primitives/test-utils/Cargo.toml:9]: And here.

File: relays/ethereum/Cargo.toml [hunk 4]
```diff
@@ -10,11 +10,11 @@ ansi_term = "0.12"
...
 clap = { version = "2.33.3", features = ["yaml"] }
-codec = { package = "parity-scale-codec", version = "1.3.4" }
+codec = { package = "parity-scale-codec", version = "2.0.0" }
 env_logger = "0.8.2"
-ethabi = "13.0"
-ethabi-contract = "11.0"
-ethabi-derive = "13.0"
+ethabi = { git = "https://github.com/paritytech/ethabi.git", branch = "td-eth-types-11" }
+ethabi-contract = { git = "https://github.com/paritytech/ethabi.git", branch = "td-eth-types-11" }
+ethabi-derive = { git = "https://github.com/paritytech/ethabi.git", branch = "td-eth-types-11" }
```
Comment (adoerr) [inline relays/ethereum/Cargo.toml:17]: So I guess, this is in order to be compatibel with the codec update, right?

File: relays/ethereum-client/Cargo.toml [hunk 5]
```diff
@@ -7,11 +7,11 @@ license = "GPL-3.0-or-later WITH Classpath-exception-2.0"
...
-codec = { package = "parity-scale-codec", version = "1.3.4" }
+codec = { package = "parity-scale-codec", version = "2.0.0" }
 headers-relay = { path = "../headers-relay" }
 hex-literal = "0.3"
 jsonrpsee = { git = "https://github.com/svyatonik/jsonrpsee.git", branch = "shared-client-in-rpc-api", default-features = false, features = ["http"] }
 libsecp256k1 = { version = "0.3.4", default-features = false, features = ["hmac"] }
 log = "0.4.11"
 relay-utils = { path = "../utils" }
-web3 = { version = "0.15", default-features = false }
+web3 = { version = "0.15", git = "https://github.com/tomusdrw/rust-web3.git", branch ="td-ethabi", default-features = false }
```
Comment (adoerr) [inline relays/ethereum-client/Cargo.toml:17]: And this would then be a consequence of the `ethabi` bump.

**Review by adoerr — APPROVED** [review]
LGTM. Might want to consider, using `finality-grandpa 0.14.0` right away. Version `0.13.0` seemed to be quite short lived.

File: modules/ethereum-contract/builtin/Cargo.toml [hunk 6]
```diff
@@ -7,9 +7,9 @@ edition = "2018"
 license = "GPL-3.0-or-later WITH Classpath-exception-2.0"
 
 [dependencies]
-codec = { package = "parity-scale-codec", version = "1.3.4" }
+codec = { package = "parity-scale-codec", version = "2.0.0" }
 ethereum-types = "0.11.0"
-finality-grandpa = "0.12.3"
+finality-grandpa = "0.13.0"
```
[the remark this answers is not in the document]
Comment (tomusdrw) [reply modules/ethereum-contract/builtin/Cargo.toml:12]: Yeah, but `substrate` is not using it yet.

File: relays/ethereum/Cargo.toml (hunk 4 above)
Comment (tomusdrw) [reply relays/ethereum/Cargo.toml:17]: Yes, upstream PR is here https://github.com/rust-ethereum/ethabi/pull/204

File: relays/ethereum-client/Cargo.toml (hunk 5 above)
Comment (tomusdrw) [reply relays/ethereum-client/Cargo.toml:17]: Indeed.

File: modules/ethereum-contract/builtin/Cargo.toml (hunk 6 above)
[the remark this answers is not in the document]
Comment (adoerr) [reply modules/ethereum-contract/builtin/Cargo.toml:12]: Seems to solved [Cargo.lock](https://github.com/paritytech/substrate/blob/e03ca38d45f438932ec92bf69a40b6b16b6ec643/Cargo.lock#L1591)

[the remark this answers is not in the document]
Comment (tomusdrw) [reply modules/ethereum-contract/builtin/Cargo.toml:12]: Right, sorry didn't make it before merge, but we can bump again soonish.

**Pull request merged**